src/main/scala/com/databricks/labs/overwatch/pipeline/Pipeline.scala

```diff
@@ -425,6 +427,33 @@ class Pipeline(
     Helpers.getDatesGlob(startDate.toLocalDate, untilTime.asLocalDateTime.plusDays(1).toLocalDate)
   }
 
+  //Get the dataframe from either tempDirectory or from the ResponseArray
+  private def getApiData(apiTempPath: Option[String]): DataFrame ={
+     //Get the data from either tempDirectory or from the ResponseArray
+     val tmpClusterEventsSuccessPath = s"${config.tempWorkingDir}/${apiTempPath.get}/success" + config.runID
```

Comment:
apiTempPath cannot be an option here and you have reference to .get which will fail if it's empty anyway. None.get -- no no

Reply:
done